### ISSUE TITLE
Example of how to make a controlled component?

### ISSUE THREAD

**Issue by ArjanJ** [rehookify/datepicker#4]
Not an issue, but some guidance would be appreciated on how to set this up so that it works like a controlled component, where the "value" (selected dates) is passed from a parent component or state, and is updated through an onChange handler of sorts. An example of how to use this with `react-hook-form` would be amazing. 

The only thing I've found so far to react to changes to `selectedDates` is by setting up a `useEffect` like this, **which honestly doesn't work very well:**

```js
const MyDatepickerComponent = ({ onChange }) => {
  const { data } = useDatePicker()
 
  useEffect(() => {
    onChange(data.selectedDates);
  }, [data.selectedDates])

  return (
    <div>...</div>
   )
}

```

What would be cool is something like this, where the `onChange` callback is called internally, and `selectedDates` is the state value from the parent component that determines what days are selected. (Making this flow of data one way/controlled):

```js
const MyDatepickerComponent = ({ onChange, selectedDates }) => {
  const { data } = useDatePicker({ dates: { onChange, selectedDates }})

  return (
    <div>...</div>
   )
}

```

Maybe there is a better way to do this, idk, any tips would be appreciated.

**Comment by Feshchenko** [comment]
Hi, Arjan.
As date-picker has multiple modes, single, multiple, and range it is hard to determine what is `onChange` means in all cases.

The first workaround is correct, you can wrap change of `selectedDates` or `formatedDates` into `useEffect`
The second workaround is when you are using propGetters you can pass `onClick` to the `dayButton`.

**Comment by ArjanJ** [comment]
Right, I would image that `onChange` would be passed `selectedDates`, that way it would work for single, multiple, or range modes. I did try passing a callback to `dayButton` but it's not that useful if you're using `multiple` or `range` modes, because then you have to create your own state management solution for tracking more than one selected date (which is what this library is already doing). I think getting the value of `selectedDates` in a more event driven way would be ideal, since it already has the value we're looking for, it's just not accessible in the right place and right time.

> About selectedDates: you could pass selectedDates to the config:

This works fine for setting the initial state, but if you update the value of `selectedDates` after the initial rendering it doesn't update. For example, clicking the "Clear" button here doesn't do anything to the state of `useDatePicker`, the initial dates are still selected.

```js
const Example = () => {
  const [myDates, setMyDates] = useState([new Date(), addDays(new Date(), 3)]);

  const dp = useDatePicker({
    dates: {
      mode: "range",
      selectedDates: myDates, // Does not update when `myDates` changes
    },
  });

  return (
    <>
      <DatePickerUI data={dp.data} propGetters={dp.propGetters} />
      <button onClick={() => setMyDates([])}>Clear dates</button>
    </>
  );
};

```

**Comment by Feshchenko** [comment]
If you need this feature urgently you can use action `setDay` https://github.com/rehookify/datepicker#setday
Personally, I don't like a controlled approach, but this is not a huge change.
I will add this to the backlog and will do it after the time picker functionality
.

**Comment by ArjanJ** [comment]
No worries, thanks for getting back to me. Not needed urgently but it would be nice to have, maybe I'll submit a PR if I get some extra time.

**Comment by Feshchenko** [comment]
Hey @ArjanJ.

I've been thinking about date-pickers and after some research, I realized that a controlled date-picker is a common pattern. So, I went ahead and made a new release, version [3.0.0](https://github.com/rehookify/datepicker/releases/tag/v3.0.0), which is fully controlled and available now on our GitHub repository.
I hope you'll find it useful and enjoy using it! Let me know if you have any feedback or questions.
